daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/SAXInfosetOutputter.scala

```diff
@@ -175,26 +180,99 @@ class SAXInfosetOutputter(xmlReader: DFDL.DaffodilParseXMLReader)
     }
   }
 
-  private def doStartElement(diElem: DIElement, contentHandler: ContentHandler): Unit = {
-    val (ns: String, elemName: String, qName: String) = getNameSpaceElemNameAndQName(diElem)
-    doStartPrefixMapping(diElem, contentHandler)
+  private def doAttributesPrefixMapping(diElem: DIElement, attrs: AttributesImpl): AttributesImpl = {
+    val (nsbStart: NamespaceBinding, nsbEnd: NamespaceBinding) = getNsbStartAndEnd(diElem)
+    var n = nsbStart
+    while (n != nsbEnd && n != null && n != scala.xml.TopScope) {
+      val prefix = if (n.prefix == null) "xmlns" else s"xmlns:${n.prefix}"
+      val uri = if (n.uri == null) "" else n.uri
+      attrs.addAttribute("", "", prefix, "CDATA", uri)
+      n = n.parent
+    }
+    attrs
+  }
 
-    val attrs = if (isNilled(diElem)) {
-      createNilAttribute()
+  private def getNsbStartAndEnd(diElem: DIElement) = {
+    val nsbStart = diElem.erd.minimizedScope
+    val nsbEnd = if (diElem.isRoot) {
+      scala.xml.TopScope
     } else {
-      new AttributesImpl()
+      diElem.diParent.erd.minimizedScope
+    }
+    (nsbStart, nsbEnd)
+  }
+
+  private def doStartElement(diElem: DIElement, contentHandler: ContentHandler): Unit = {
+    val (ns: String, localName: String, qName: String) = getNamespaceLocalNameAndQName(diElem)
+    val attrs = new AttributesImpl()
+    var elemUri: String = ""
+    var elemLocalName: String = ""
+    var elemQname: String = ""
+
+    if (namespacesFeature) {
```

Comment:
I think a more scala-y way to do this and avoid the var's is something like:
```scala
val (elemUri, elemLocalName) =
  if (namespacesFeature) {
    ...
    (ns, localName)
  } else {
    ("", "")
  }
```
Same thing for elemQName below.